File: 1175-array-transformation/1175-array-transformation.java
class Solution {
    public List<Integer> transformArray(int[] arr) {
        int[] answer = arr.clone();
        
        for (int i = 1; i < arr.length - 1; ++i) {
            if (arr[i-1] < arr[i] && arr[i] > arr[i+1])
                answer[i] -= 1;
            if (arr[i-1] > arr[i] && arr[i] < arr[i+1])
                answer[i] += 1;
        }
        
        return Arrays.equals(arr, answer) ? Arrays.stream(answer).boxed().toList() : transformArray(answer);
    }
}

File: 1175-array-transformation/1175-array-transformation.py
class Solution:
    def transformArray(self, arr: List[int]) -> List[int]:
        n = len(arr)
        answer = arr.copy()
        
        for i in range(1, n-1):
            if arr[i-1] > arr[i] < arr[i+1]:
                answer[i] += 1
            elif arr[i-1] < arr[i] > arr[i+1]:
                answer[i] -= 1

        return answer if arr == answer else self.transformArray(answer)